src/plugins/data/public/index_patterns/index.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { errors } from './errors';
+import {
+  ILLEGAL_CHARACTERS,
+  ILLEGAL_CHARACTERS_KEY,
+  CONTAINS_SPACES_KEY,
+  ILLEGAL_CHARACTERS_VISIBLE,
+  validate,
+  getFromSavedObject,
+  getRoutes,
+  findByTitle,
+} from './utils';
+
+export const IndexPattern = {
```

Comment:
Please rename this to esIndexPattern for now.
`IndexPattern` is a name already used for the class.

Reply:
I think `esIndexPattern` is going to be confusing because index patterns in ES are a different concept from index patterns in kibana.

I'd vote to use the same name we use for the service when namespacing static exports related to that service, e.g.
```ts
// if stateful runtime dependencies live in `indexPatterns`...
data.indexPatterns.foo();

// ...i'd expect static code to use the same name
import { indexPatterns } from 'src/plugins/data/public';
indexPatterns.bar();
```

Reply:
Fine by me :-)